trio/_core/_asyncgens.py

```diff
@@ -22,17 +35,17 @@ class AsyncGenerators:
     # asyncgens after the system nursery has been closed, it's a
     # regular set so we don't have to deal with GC firing at
     # unexpected times.
-    alive = attr.ib(factory=weakref.WeakSet)
+    alive: _WEAK_ASYNC_GEN_SET | _ASYNC_GEN_SET = attr.ib(factory=_WEAK_ASYNC_GEN_SET)
 
     # This collects async generators that get garbage collected during
     # the one-tick window between the system nursery closing and the
     # init task starting end-of-run asyncgen finalization.
-    trailing_needs_finalize = attr.ib(factory=set)
+    trailing_needs_finalize: _ASYNC_GEN_SET = attr.ib(factory=_ASYNC_GEN_SET)
 
     prev_hooks = attr.ib(init=False)
 
-    def install_hooks(self, runner):
-        def firstiter(agen):
+    def install_hooks(self, runner: _run.Runner) -> None:
+        def firstiter(agen: AsyncGeneratorType[object, NoReturn]) -> None:
```

Comment:
firstiter and finalizer in typeshed are typed as taking `collections.abc.AsyncGenerator[Any, Any]`. I think it would be better for us to use that name and avoid the need for a type-ignore. Using Any is not evil.